### DISCUSSION TITLE
望deepin可以做的轻量化一些或增加自定义功能

### DISCUSSION THREAD

**Discussion by jidro** [linuxdeepin/developer-center#3262] [Features Request & Ideas | 特性请求 & 头脑风暴]
问题：deepin有些臃肿，希望可以轻量化一些或者自己选择安装哪些。
建议一：在设置或者高级设置里面增加一个可以选择下载插件（或者说组件）或者删除组件的选项，自己需要的自己选择安装，不需要的自己选择删除。
建议二：安装deepin系统的时候自定义安装也可以选择自己需要哪些，最终安装哪些，不懂的可以全盘安装或者自定义安装的时候有个默认配置（默认配置是安装全部组件）。

**Comment by BLumia** [comment]
如果指的是安装器安装后的情况的话，其实安装速度和组件化安装是有取舍的。目前为了安装速度上有保障，其实是类似解压性质直接把一个预先准备好的环境写进去的，固然也可以做到先这样准备一个最小环境，但后面安装其它组件就会更加耗时了。

后续不清楚有没有实际去做组件化安装的打算， cc @justforlxz @meiyixiang

**Reply by justforlxz** [reply]
有热心的小伙伴打算一块来写一个基于网络安装的安装器，但是组件安装其实是需要人工分类数据的，这个不太确定能不能维护下去，按照之前做服务器版的经验，太容易翻车了。

**Reply by jidro** [reply]
对，是安装器，但是不仅限于安装器，还可以在系统安装完成之后选择增加或减少某些模块。
这件事灵感是：
①之前我用老电脑安装deepin发现蓝牙不能用了，蓝牙在windows上是正常的，但是deepin就没法用，想着删除设置里面的蓝牙模块，还不知道咋删咋卸载。
②有时候手动更新系统设置里的网络模块会莫名消失，但是网络是正常已经连接好的，无线网和有线网都可以自动连接，只是设置里面找不到网络模块了，只能重装系统，重装之后就有了。
③新购买的电脑没有摄像头，也没有指纹识别，这俩在设置里面对我来说根本没有用，也没法卸载，只能干瞪眼。
所以想着可以增加一个功能，无论是全新安装系统，还是已经安装好了，都可以安装或者卸载某些自己觉得没用的模块。

**Comment by meiyixiang** [comment]
专业版应该是没有这个计划的，也没有这个需求，如果要做应该就是我们自己做或者发动社区小伙伴。

**Comment by jidro** [comment]
对，是安装器，但是不仅限于安装器，还可以在系统安装完成之后选择增加或减少某些模块。
这件事灵感是：
①之前我用老电脑安装deepin发现蓝牙不能用了，蓝牙在windows上是正常的，但是deepin就没法用，想着删除设置里面的蓝牙模块，还不知道咋删咋卸载。
②有时候手动更新系统设置里的网络模块会莫名消失，但是网络是正常已经连接好的，无线网和有线网都可以自动连接，只是设置里面找不到网络模块了，只能重装系统，重装之后就有了。
③新购买的电脑没有摄像头，也没有指纹识别，这俩在设置里面对我来说根本没有用，也没法卸载，只能干瞪眼。
所以想着可以增加一个功能，无论是全新安装系统，还是已经安装好了，都可以安装或者卸载某些自己觉得没用的模块。

**Comment by meiyixiang** [comment]
谢谢你的反馈，这个问题我记录并放入需求池了。我觉得他是个不错的需求，但可能工作量比较大，我们会排期去考虑何时去实现这类比较灵活的个性化定制需求。
